File: SPCServer/springboot/src/main/java/com/lcaohoanq/shoppe/services/auth/IAuthService.java
package com.lcaohoanq.shoppe.services.auth;

import com.lcaohoanq.shoppe.dtos.request.AccountRegisterDTO;
import com.lcaohoanq.shoppe.dtos.responses.UserResponse;
import com.lcaohoanq.shoppe.models.User;

public interface IAuthService {

    User register(AccountRegisterDTO accountRegisterDTO) throws Exception;
    String login(String email, String password) throws Exception;
    UserResponse getUserDetailsFromToken(String token) throws Exception;
    void logout(String token, User user) throws Exception;
    void verifyOtpToVerifyUser(Long userId, String otp) throws Exception;
    void verifyOtpIsCorrect(Long userId, String otp) throws Exception;
}

File: auth-service/src/main/kotlin/com/lcaohoanq/authservice/domains/auth/IAuthService.kt
package com.lcaohoanq.authservice.domains.auth

import com.lcaohoanq.authservice.domains.user.User
import com.lcaohoanq.common.dto.TokenPort
import com.lcaohoanq.authservice.domains.user.UserPort
import com.lcaohoanq.common.dto.AuthPort


interface IAuthService {

    fun login(account: AuthPort.AuthRequest): LoginResult
    fun register(newAccount: AuthPort.SignUpReq)
    fun getUserDetailsFromToken(token: String): UserPort.UserResponse
    fun getCurrentAuthenticatedUser(): User
    fun refreshToken(refreshTokenDTO: TokenPort.RefreshTokenDTO): AuthPort.AuthResponse
    fun logout(token: String, user: User)
    fun generateTokenFromEmail(email: String): String
    fun changePassword(req: AuthPort.ChangePasswordReq)
    fun verifyAccount(token: String): Unit
    fun disableAccount(token: String)
    fun setup2FA(email: String): String
    fun verify2fa(data: AuthPort.Verify2FAReq)

}